### DISCUSSION TITLE
Is the GMS version more efficient than OSS?

### DISCUSSION THREAD

**Discussion by 1RandomDev** [owntracks/android#1276] [Q&A] [answered]
Hey, I recently used the OSS version of OwnTracks for some time and had issues with significant battery drain.
So does "not using any Google APIs" also mean that it can't use the Network Location API of microG/UnifiedNLP and so has to use GPS for every location update, which obviously would explain the high battery usage?

**Comment by growse** [comment, marked as answer]
Yes.

The GMS build uses the [Fused Location Provider](https://developers.google.com/location-context/fused-location-provider) which is a lot more battery efficient than just polling GPS (which is what the OSS version does). Currently, it uses the `GpsMyLocationProvider` from [OSMDroid](https://osmdroid.github.io/osmdroid/javadocAll/org/osmdroid/views/overlay/mylocation/GpsMyLocationProvider.html).

I'm sure there's probably a better way of doing this, but don't have any good ideas at the moment.

**Reply by 1RandomDev** [reply]
Ok so if I use the GMS version with microG and UnifiedNLP is there any tracking by Google like Firebase Analytics and all this shit?

**Reply by growse** [reply]
Disclaimer: I'm not an expert in microg and how it works.

However, my understanding is that it provides an implementation of the Play Services API used by the various closed source Google Android libraries (play-services-maps etc.) that doesn't talk to Google.

*If* this is true, and *if* the android client libs don't make network requests to Google themselves (Vs delegating to the play services framework), then it seems like running the GMS version on microg shouldn't send any data to Google....? You can check the dependencies that only the GMS version pulls in here: https://github.com/owntracks/android/blob/bb280f33fbf4414cf6f4b39e4592ab11a36add89/project/app/build.gradle.kts#L159-164

This should be a somewhat straightforward thing to test - if you get a microg device, load the OT GMS flavour on, and capture the network traffic (tcpdump on the router), you should be able to build up a picture of what IP/ports traffic is flowing to. 

(I'm not endorsing/recommending the microg approach here, the Right Solution to this is for the OSS build to give as good a battery experience as the GMS flavour, but I also know it's a pragmatic option in the interim).

**Reply by 1RandomDev** [reply]
Thx network requests look good so far, only my server and the selected map provider. And the battery usage is now almost not noticeable.
